src/blob_store.rs

```diff
@@ -73,43 +69,57 @@ impl<V: Blob> BlobStore<V> {
     ///
     /// `base_path` is the directory where the storage files will be stored.
     /// It should exist already.
-    pub fn new(base_path: PathBuf, page_size: Option<usize>) -> Self {
-        assert!(base_path.exists(), "Base path does not exist");
-        assert!(base_path.is_dir(), "Base path is not a directory");
-        let page_size = page_size.unwrap_or(PAGE_SIZE_BYTES);
+    pub fn new(base_path: PathBuf, options: StorageOptions) -> Result<Self, &'static str> {
+        if !base_path.exists() {
+            return Err("Base path does not exist");
+        }
+        if !base_path.is_dir() {
+            return Err("Base path is not a directory");
+        }
+
+        let config = StorageConfig::try_from(options)?;
+
+        // write config to disk
+        let config_path = base_path.join(CONFIG_FILENAME);
+        let file = std::fs::File::create(&config_path).unwrap();
+        serde_json::to_writer_pretty(file, &config).unwrap();
```

Comment:
also maybe better to save last as a signal that everything has been done before

then opening storage, look at storage first